ledger/acctupdates_test.go

```diff
@@ -1030,3 +1032,89 @@ func TestListCreatables(t *testing.T) {
 		expectedDbImage, numElementsPerSegement)
 	listAndCompareComb(t, au, expectedDbImage)
 }
+
+func TestIsWritingCatchpointFile(t *testing.T) {
+
+	au := &accountUpdates{}
+
+	au.catchpointWriting = make(chan struct{}, 1)
+	ans := au.IsWritingCatchpointFile()
+	require.True(t, ans)
+
+	close(au.catchpointWriting)
+	ans = au.IsWritingCatchpointFile()
+	require.False(t, ans)
+}
+
+func TestGetCatchpointStream(t *testing.T) {
+
+	proto := config.Consensus[protocol.ConsensusCurrentVersion]
+
+	ml := makeMockLedgerForTracker(t, true)
+	defer ml.close()
+	ml.blocks = randomInitChain(protocol.ConsensusCurrentVersion, 10)
+
+	accts := []map[basics.Address]basics.AccountData{randomAccounts(20, true)}
+	au := &accountUpdates{}
+	conf := config.GetDefaultLocal()
+	conf.CatchpointInterval = 1
+	au.initialize(conf, ".", proto, accts[0])
+	defer au.close()
+
+	err := au.loadFromDisk(ml)
+	require.NoError(t, err)
+
+	filesToCreate := 4
+
+	temporaryDirectroy, err := ioutil.TempDir(os.TempDir(), "catchpoints")
+	require.NoError(t, err)
+	defer func() {
+		os.RemoveAll(temporaryDirectroy)
+	}()
+	catchpointsDirectory := filepath.Join(temporaryDirectroy, "catchpoints")
+	err = os.Mkdir(catchpointsDirectory, 0777)
+	require.NoError(t, err)
+
+	au.dbDirectory = temporaryDirectroy
+
+	// Create the catchpoint files with dummy data
+	for i := 0; i < filesToCreate; i++ {
+		fileName := filepath.Join("catchpoints", fmt.Sprintf("%d.catchpoint", i))
```

Comment:
join with `catchpointsDirectory` here to avoid the subsequent `filepath.Join(temporaryDirectroy, fileName)`

Reply:
I will need fileName without temoraryDirectory on line 1088.
The reason why I need this arrangement is to match the filename in GetCatchpointStream:
```
	// if the database doesn't know about that round, see if we have that file anyway:
	fileName := filepath.Join("catchpoints", catchpointRoundToPath(round))
	catchpointPath := filepath.Join(au.dbDirectory, fileName)

```